utils/utils.go

```diff
@@ -32,12 +32,16 @@ func sourceDir(file string) string {
 
 // FileWithLineNum return the file name and line number of the current file
 func FileWithLineNum() string {
-	// the second caller usually from gorm internal, so set i start from 2
-	for i := 2; i < 15; i++ {
-		_, file, line, ok := runtime.Caller(i)
```

Comment:
The `runtime.Caller(i)` 's inner is `caller(skip+1)`, so we use `runtime.Callers(3)` in new implement.